File: JavaScript/7js 1/args_count.js
/*
|--------------------------------------------------------------------------
| How many arguments
|--------------------------------------------------------------------------
|
*/

function args_count() {
   return arguments.length
}


function args_count(...args) {
   return args.length
}


function args_count() {
   let i = 0;
   for (argument of arguments) {
      i++
   }
   return i;
}


console.log(args_count(1, 2), 2);
console.log(args_count(), 0);
console.log(args_count('A', 'B', 'C'), 3);
console.log(args_count(["foo", "bar"]), 1);

File: Python/7py 1/args_count.py
"""
|--------------------------------------------------------------------------
| How many arguments
|--------------------------------------------------------------------------
|
"""

def args_count(*args, **kwargs):
   return len(args) + len(kwargs)


def args_count(*args, **kwargs):
   res = 0
   for x in args:
      res += 1
   for y in kwargs:
      res += 1
   return res


def args_count(*args, **kwargs):
   return len([*args, *kwargs])


print(args_count(100), 1)
print(args_count(100, 2, 3), 3)
print(args_count(32, a1=12), 2)
print(args_count(), 0)